sklearn/tests/test_multioutput.py

```diff
@@ -336,3 +344,160 @@ def test_multi_output_exceptions():
+
+    Y_pred_binary = (Y_decision >= 0)
+    assert_greater(jaccard_similarity_score(Y, Y_pred_binary), 0.4)
+
+
+def test_classifier_chain_fit_and_predict_with_sparse_data():
+    """Fit classifier chain with sparse data"""
+    X, Y = generate_multilabel_dataset_with_correlations()
+
+    X_sparse = sp.csr_matrix(X)
+    classifier_chain = ClassifierChain(LogisticRegression())
+    classifier_chain.fit(X_sparse, Y)
+    Y_pred = classifier_chain.predict(X_sparse)
+    assert_equal(Y_pred.shape, Y.shape)
+
+
+def test_classifier_chain_fit_and_predict_with_sparse_data_and_cv():
+    """Fit classifier chain with sparse data cross_val_predict"""
+    X, Y = generate_multilabel_dataset_with_correlations()
+    X_sparse = sp.csr_matrix(X)
+    classifier_chain = ClassifierChain(LogisticRegression(), cv=3)
+    classifier_chain.fit(X_sparse, Y)
+    Y_pred = classifier_chain.predict(X_sparse)
+    assert_equal(Y_pred.shape, Y.shape)
+
+
+def test_classifier_chain_random_order():
+    """Fit classifier chain with random order"""
+    X, Y = generate_multilabel_dataset_with_correlations()
+    classifier_chain = ClassifierChain(LogisticRegression(),
+                                       order='random')
+    classifier_chain.fit(X, Y)
+    assert_not_equal(list(classifier_chain.order), list(range(4)))
+    assert_equal(len(classifier_chain.order), 4)
+    assert_equal(len(set(classifier_chain.order)), 4)
+
+
+def test_classifier_chain_parameter_and_output_sizes():
+    """Test classifier chain parameter and output sizes"""
+
+    X, Y = generate_multilabel_dataset_with_correlations()
+    classifier_chain = ClassifierChain(LogisticRegression())
+    classifier_chain.fit(X, Y)
+
+    Y_pred = classifier_chain.predict(X)
+    assert_equal(Y_pred.shape, Y.shape)
+
+    Y_prob = classifier_chain.predict_proba(X)
+    assert_equal(Y_prob.shape, Y.shape)
+
+    Y_decision = classifier_chain.decision_function(X)
+    assert_equal(Y_decision.shape, Y.shape)
+
+    assert_equal([c.coef_.size for c in classifier_chain.estimators_],
+                 list(range(X.shape[1], X.shape[1] + Y.shape[1])))
+
+
+def test_classifier_chain_crossval_fit_and_predict():
+    """Fit classifier chain with cross_val_predict and verify predict
+    performance"""
+    X, Y = generate_multilabel_dataset_with_correlations()
+    classifier_chain_cv = ClassifierChain(LogisticRegression(), cv=3)
+    classifier_chain_cv.fit(X, Y)
+
+    classifier_chain = ClassifierChain(LogisticRegression())
+    classifier_chain.fit(X, Y)
+
+    Y_pred_cv = classifier_chain_cv.predict(X)
+    Y_pred = classifier_chain.predict(X)
+
+    assert_equal(Y_pred_cv.shape, Y.shape)
+    assert_greater(jaccard_similarity_score(Y, Y_pred_cv), 0.4)
+
+    assert_not_equal(jaccard_similarity_score(Y, Y_pred_cv),
+                     jaccard_similarity_score(Y, Y_pred))
+
+
+def test_classifier_chain_vs_independent_models():
+    """Verify that an ensemble of classifier chains (each of length
```

Comment:
due to issues with nose, we don't usually use docstrings in test functions. Comments are preferred